packages/sdk/src/types/runtime/behaviors/action.ts

```diff
@@ -24,13 +30,31 @@ export class DiscreteAction {
 	private handlers: ActionHandlers = {};
 	private activeUserIds: string[] = [];
 
+	/** @hidden */
+	constructor(private name: string) {
+	}
+
 	/**
 	 * Add a handler for the given action state for when it is triggered.
 	 * @param actionState The action state that the handle should be assigned to.
 	 * @param handler The handler to call when the action state is triggered.
 	 */
-	public on(actionState: ActionState, handler: ActionHandler): this {
+	public on(
+		context: Context,
+		actorId: string,
+		actionState: ActionState,
+		options: ActionHandler | ActionHandlerWithTriggeredAction): this {
+		const handler = (typeof options === 'function') ? options : options.handler;
 		this.handlers[actionState] = handler;
+		if (typeof options === 'object') {
+			context.internal.sendPayload({
+				type: 'set-triggered-action',
+				targetId: actorId,
+				actionName: this.name,
+				actionState,
+				action: options.action,

```

Comment:
>action [](start = 4, length = 6)

I suppose this should be called `triggeredAction` to differentiate meaning from `actionName` and `actionState`, which use the term "action" differently.

Reply:
I agree.